src/dso_api/dynamic_api/temporal.py

```diff
@@ -0,0 +1,144 @@
+
+    def __bool__(self):
+        return self.is_versioned
+
+    def __init__(self, request, table_schema: DatasetTableSchema):
+        temporal = table_schema.temporal
+
+        self.table_schema = table_schema
+        self.is_versioned = temporal is not None
+
+        # Parsed arguments are None unless there is a match with the request
+        self.version_field = None
+        self.version_value = None
+
+        self.slice_dimension = None
+        self.slice_value = None
+        self.slice_range_fields = None
+
+        if temporal is not None:
+            # See if a filter is made on a specific version
+            self.version_field = temporal.identifier  # e.g. "volgnummer"
+            if version := request.GET.get(self.version_field):
+                self.version_value = version
+
+            # See if a filter is done on a particular point in time (e.g. ?geldigOp=yyyy-mm-dd)
+            for dimension, fields in temporal.dimensions.items():
+                if date_value := request.GET.get(dimension):
+                    self.slice_dimension = dimension  # e.g. geldigOp
+                    self.slice_value = self._parse_date(dimension, date_value)
+                    self.slice_range_fields = fields
+
+    def _parse_date(self, dimension: str, value: str) -> Union[Literal["*"], date, datetime]:
+        """Parse and validate the received date"""
+        if value == "*":
+            return "*"
+
+        try:
+            if "T" in value or " " in value:
+                return make_aware(datetime.fromisoformat(value))
+            else:
+                return date.fromisoformat(value)
+        except ValueError:
+            raise ValidationError(
+                f"Invalid date or date-time format for '{dimension}' parameter!"
+            ) from None
+
+    def filter_object_version(self, queryset: models.QuerySet) -> models.QuerySet:
+        """Apply an object-level filter such as "?volgnummer=...".
+        This should only be called for the top-level object.
+        """
+        if queryset.model.table_schema() != self.table_schema:
+            raise ValueError("QuerySet model type does not match")
+
+        if not self.is_versioned:
+            return queryset
+        elif self.version_value is None:
+            return self.filter_queryset(queryset)
+        else:
+            return queryset.filter(**{self.version_field: self.version_value})
+
+    def filter_queryset(self, queryset: models.QuerySet) -> models.QuerySet:
+        if queryset.model.table_schema() != self.table_schema:
+            raise ValueError("QuerySet model type does not match")
+
+        if not self.is_versioned or self.slice_value == "*":
+            # allow this method to be called unconditionally on objects,
+            # and allow ?geldigOp=* to return all data
+            return queryset
+
+        # Either take a given ?geldigOp=yyyy-mm-dd OR ?geldigOp=NOW()
+        slice_value = self.slice_value or now()
```

Comment:
Elsewhere you are making sure you are dealing with aware `datetime` objects. But here you are using a naive `datetime` object (as returned by `now()`. Why does it not matter here?

Reply:
I'm using `django.utils.timezone.now()` which is aware.